ExecuteScriptRequestTest.testExecuteScriptRequest_whenTargetAllMembers & testExecuteScriptRequest_whenTargetAllMembers_withTarget

**testExecuteScriptRequest_whenTargetAllMembers**
```
expected:<[error
error]
> but was:<[java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: Could not find ScriptEngine named 'JavaScript'.
 at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrow(InvocationFuture.java:85)
 at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:150)
 at com.hazelcast.internal.management.ManagementCenterService.callOnAddress(ManagementCenterService.java:247)
 at com.hazelcast.internal.management.ManagementCenterService.callOnMember(ManagementCenterService.java:259)
 at com.hazelcast.internal.management.request.ExecuteScriptRequest.writeResponse(ExecuteScriptRequest.java:83)
 at com.hazelcast.internal.management.ExecuteScriptRequestTest.testExecuteScriptRequest_whenTargetAllMembers(ExecuteScriptRequestTest.java:64)
 at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
 at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 at java.lang.reflect.Method.invoke(Method.java:622)
 at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
 at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
 at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
 at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
 at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:88)
 at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:82)
 at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:334)
 at java.util.concurrent.FutureTask.run(FutureTask.java:166)
 at java.lang.Thread.run(Thread.java:701)
Caused by: java.lang.IllegalArgumentException: Could not find ScriptEngine named 'JavaScript'.
 at com.hazelcast.internal.management.operation.ScriptExecutorOperation.run(ScriptExecutorOperation.java:57)
 at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:181)
 at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:375)
 at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:402)
 at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:283)
 at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:268)
 at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:232)
 at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:207)
 at com.hazelcast.spi.impl.operationservice.impl.OperationServiceImpl.invokeOnTarget(OperationServiceImpl.java:310)
 at com.hazelcast.internal.management.ManagementCenterService.callOnAddress(ManagementCenterService.java:245)
 at com.hazelcast.internal.management.ManagementCenterService.callOnMember(ManagementCenterService.java:259)
 at com.hazelcast.internal.management.request.ExecuteScriptRequest.writeResponse(ExecuteScriptRequest.java:83)
 at com.hazelcast.internal.management.ExecuteScriptRequestTest.testExecuteScriptRequest_whenTargetAllMembers(ExecuteScriptRequestTest.java:64)
 at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
 at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 at java.lang.reflect.Method.invoke(Method.java:622)
 at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
 at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
 at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
 at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
 at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:88)
 at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:82)
 at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:334)
 at java.util.concurrent.FutureTask.run(FutureTask.java:166)
 at java.lang.Thread.run(Thread.java:701)
 at ------ submitted from ------.(Unknown Source)
 at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolve(InvocationFuture.java:111)
 at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrow(InvocationFuture.java:74)
 ... 18 more

java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: Could not find ScriptEngine named 'JavaScript'.
 at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrow(InvocationFuture.java:85)
 at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:158)
 at com.hazelcast.internal.management.ManagementCenterService.callOnAddress(ManagementCenterService.java:247)
 at com.hazelcast.internal.management.ManagementCenterService.callOnMember(ManagementCenterService.java:259)
 at com.hazelcast.internal.management.request.ExecuteScriptRequest.writeResponse(ExecuteScriptRequest.java:83)
 at com.hazelcast.internal.management.ExecuteScriptRequestTest.testExecuteScriptRequest_whenTargetAllMembers(ExecuteScriptRequestTest.java:64)
 at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
 at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 at java.lang.reflect.Method.invoke(Method.java:622)
 at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
 at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
 at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
 at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
 at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:88)
 at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:82)
 at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:334)
 at java.util.concurrent.FutureTask.run(FutureTask.java:166)
 at java.lang.Thread.run(Thread.java:701)
Caused by: java.lang.IllegalArgumentException: Could not find ScriptEngine named 'JavaScript'.
 at com.hazelcast.internal.management.operation.ScriptExecutorOperation.run(ScriptExecutorOperation.java:57)
 at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:181)
 at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:396)
 at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:117)
 at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.run(OperationThread.java:102)
 at ------ submitted from ------.(Unknown Source)
 at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolve(InvocationFuture.java:111)
 at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrow(InvocationFuture.java:74)
 ... 18 more
]
>
```

**testExecuteScriptRequest_whenTargetAllMembers_withTarget**
```
expected:<[error]> but was:<[java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: Could not find ScriptEngine named 'JavaScript'.
 at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrow(InvocationFuture.java:85)
 at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:150)
 at com.hazelcast.internal.management.ManagementCenterService.callOnAddress(ManagementCenterService.java:247)
 at com.hazelcast.internal.management.request.ExecuteScriptRequest.writeResponse(ExecuteScriptRequest.java:91)
 at com.hazelcast.internal.management.ExecuteScriptRequestTest.testExecuteScriptRequest_whenTargetAllMembers_withTarget(ExecuteScriptRequestTest.java:78)
 at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
 at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 at java.lang.reflect.Method.invoke(Method.java:622)
 at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
 at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
 at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
 at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
 at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:88)
 at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:82)
 at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:334)
 at java.util.concurrent.FutureTask.run(FutureTask.java:166)
 at java.lang.Thread.run(Thread.java:701)
Caused by: java.lang.IllegalArgumentException: Could not find ScriptEngine named 'JavaScript'.
 at com.hazelcast.internal.management.operation.ScriptExecutorOperation.run(ScriptExecutorOperation.java:57)
 at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:181)
 at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:375)
 at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:402)
 at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:283)
 at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:268)
 at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:232)
 at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:207)
 at com.hazelcast.spi.impl.operationservice.impl.OperationServiceImpl.invokeOnTarget(OperationServiceImpl.java:310)
 at com.hazelcast.internal.management.ManagementCenterService.callOnAddress(ManagementCenterService.java:245)
 at com.hazelcast.internal.management.request.ExecuteScriptRequest.writeResponse(ExecuteScriptRequest.java:91)
 at com.hazelcast.internal.management.ExecuteScriptRequestTest.testExecuteScriptRequest_whenTargetAllMembers_withTarget(ExecuteScriptRequestTest.java:78)
 at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
 at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 at java.lang.reflect.Method.invoke(Method.java:622)
 at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
 at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
 at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
 at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
 at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:88)
 at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:82)
 at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:334)
 at java.util.concurrent.FutureTask.run(FutureTask.java:166)
 at java.lang.Thread.run(Thread.java:701)
 at ------ submitted from ------.(Unknown Source)
 at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolve(InvocationFuture.java:111)
 at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrow(InvocationFuture.java:74)
 ... 17 more]>
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x-ZuluJDK6/330/
https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x-ZuluJDK7/334/

--- a/hazelcast/src/test/java/com/hazelcast/internal/management/ExecuteScriptRequestTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/internal/management/ExecuteScriptRequestTest.java
@@ -11,6 +11,7 @@ import com.hazelcast.test.TestHazelcastInstanceFactory;
 import com.hazelcast.test.annotation.ParallelTest;
 import com.hazelcast.test.annotation.QuickTest;
 import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
@@ -32,6 +33,13 @@ public class ExecuteScriptRequestTest extends HazelcastTestSupport {
     private Cluster cluster;
     private ManagementCenterService managementCenterService;
     private Map<String, Object> bindings = new HashMap<String, Object>();
+
+    /**
+     * Zulu 6 and 7 doesn't have Rhino script engine, so this test should be excluded.
+     * see http://zulu.org/forum/thread/nullpointerexception-on-loading-the-javascript-engine-in-zulu-7-u76/
+     */
+    @Rule
+    public ZuluExcludeRule zuluExcludeRule = new ZuluExcludeRule();
 
     @Before
     public void setUp() {

--- a/hazelcast/src/test/java/com/hazelcast/internal/management/ZuluExcludeRule.java
+++ b/hazelcast/src/test/java/com/hazelcast/internal/management/ZuluExcludeRule.java
@@ -1,0 +1,54 @@
+package com.hazelcast.internal.management;
+
+import org.junit.rules.MethodRule;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.Statement;
+
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Exclusion rule for Zulu JDK 6 and 7.
+ */
+public class ZuluExcludeRule implements MethodRule {
+
+    private static final boolean EXCLUDED;
+
+    static {
+        String propertyName = "java.specification.version";
+        String versionProperty = System.getProperty(propertyName);
+        assertNotNull(propertyName + " should be set!", versionProperty);
+
+        String vendorPropertyName = "java.vm.vendor";
+        String vendor = System.getProperty(vendorPropertyName);
+        assertNotNull(vendorPropertyName + " should be set!", vendor);
+
+        int version = Integer.parseInt(versionProperty.split("\\.")[1]);
+        EXCLUDED = version < 8 && vendor.startsWith("Azul");
+    }
+
+    @Override
+    public Statement apply(Statement statement, FrameworkMethod frameworkMethod, Object o) {
+        if (EXCLUDED) {
+            return new ExcludedStatement(frameworkMethod);
+        } else {
+            return statement;
+        }
+    }
+
+    private static final class ExcludedStatement extends Statement {
+
+        private final FrameworkMethod method;
+
+        private ExcludedStatement(FrameworkMethod method) {
+            this.method = method;
+        }
+
+        @Override
+        public void evaluate() throws Throwable {
+            String className = method.getMethod().getDeclaringClass().getName();
+            String methodName = method.getName();
+
+            System.out.println("EXCLUDING '" + className + "#" + methodName + "()' ...");
+        }
+    }
+}

